### PR TITLE
Add NLP problem LINCON

### PR DESCRIPTION
**Pull request by amontoison**
Implementation of the model used in `NLPModelsJuMP`.

### CODE REVIEW

File: test/consistency.jl [hunk 1]
```diff
@@ -413,7 +413,7 @@ function consistent_nlps(nlps; rtol=1.0e-8)
   # If there are inequalities, test the SlackModels of each of these models
   if nlps[1].meta.ncon > length(nlps[1].meta.jfix)
     slack_nlps = [SlackModel(nlp) for nlp in nlps]
-    consistent_functions(slack_nlps)
+    consistent_functions(slack_nlps, exclude=[cons])
```
Comment (amontoison) [inline test/consistency.jl]: With a `SlackModel`, lower bounds and upper bounds of constraints become lower / upper bounds of slack variables. So if both models are different constraints bounds, I cannot compare `cons` functions.

With linear constraints and a `MathOptNLPModel`, `l <= Ax + b <= u` is internally stored as `l - b <= Ax <= u - b`.

File: test/problems/lincon.jl [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+function NLPModels.grad!(nlp :: LINCON, x :: AbstractVector, gx :: AbstractVector)
+  increment!(nlp, :neval_grad)
+  gx .= [4 * x[i]^3 for i =1:nlp.meta.nvar]
+  return gx
+end
+
+function NLPModels.hess(nlp :: LINCON, x :: AbstractVector{T}; obj_weight=one(T)) where T
+  increment!(nlp, :neval_hess)
+  return T.(diagm([12.0 * x[i]^2 for i=1:nlp.meta.nvar])) * obj_weight
```
Comment (abelsiqueira) [inline test/problems/lincon.jl]: I think you can change `12.0` to `12` and drop `T`

File: test/consistency.jl (hunk 1 above)
Comment (abelsiqueira) [reply test/consistency.jl]: Can you change the example here in `NLPModels` to `l - b <= Ax <= u - b`?

**Comment by abelsiqueira** [comment]
Thank you

**Pull request merged**